tests/rptest/infinite_retention/checks.py

```diff
@@ -0,0 +1,156 @@
+            names2const = NamesToConstants(self.args)
+            # iterate all checks
+            for idx in range(len(self.checks)):
+                # get expression for this check
+                _item = self.checks[idx]
+                _expr = _item['expr']
+                # Do the magic:
+                # - 'precompile/parse' it
+                # - visit each argument/node and update to values
+                # - unparse it back to text
+                _parsed = parse(_expr)
+                names2const.visit(_parsed)
+                _item['expr_text'] = unparse(_parsed).strip()
+                # evaluate text version of expression
+                # using locally stored arguments
+                _item['result'] = eval(_item['expr_text'], {}, self.args)
+            # Mark that we calculated all expressions
+            self._calculated = True
+
+    # Create text Summary to show in logs
+    def get_summary_as_text(self):
+        def format_items(iterator):
+            _fmt = "\n{0}\n-> {1}: {2}\n"
+            _str = ""
+            for item in iterator:
+                # Show in summary only if result is false or show is true
+                if item["show"] or not item['result']:
+                    _str += _fmt.format(item['expr'], item['result'],
+                                        item['expr_text'])
+            return _str
+
+        if not self.summary:
+            _summary = "\n# Summary:"
+
+            _failed = filter(lambda x: not x['result'], self.checks)
+            _failed_summary = format_items(_failed)
+            if len(_failed_summary) > 0:
+                _summary += "\n## Failed checks:"
+                _summary += _failed_summary
+
+            _passed = filter(lambda x: x['result'], self.checks)
+            _passed_summary = format_items(_passed)
+            if len(_passed_summary) > 0:
+                _summary += "\n## Passed checks:"
+                _summary += _passed_summary
+
+            self.summary = _summary
+        return self.summary
+
+    # Assert calculated results and generate AssertionError
+    # if any fails occured
+    def assert_results(self):
+        _r = filter(lambda x: not x['result'], self.checks)
+        if len(list(_r)) > 0:
+            raise AssertionError("At least one check failed.\n" +
+                                 self.get_summary_as_text())
+        else:
+            return True
+
+
+def main():
+    # TechDemo:
+    # args = {"a": 5, "b": 6}
+    # expr = "a+b*2"
+    # parsed_expr = parse(expr)
+    # NamesToConstants(args).visit(parsed_expr)
+    # expr_values = unparse(parsed_expr).strip()
+    # result = eval("a+b*2", {}, args)
+    # print(f"Simple check: {expr}")
+    # print(f"{expr_values} = {result}")
+
+    # this is very simple unittest for Checks :)
+    chk = RedpandaChecksBase()
+    chk.add_arg("value1", 23)
+    chk.add_arg("value2", 56)
+    chk.add_check("Simple formula", "value1 <= value2")
```

Comment:
I'm curious about the benefit of using a string of python code here, as opposed to a lambda: I could imagine expressing this as something like `add_check("Something", lambda args: args.value1 <= args.value2)`, which would avoid the need to do do runtime eval + let your editor highlight the code etc.

Not objecting, just curious.

Reply:
String passed to eval and in combination with `NamesToConstants` this evaluates to the output that shows formula AND formula with actual calculated values during the test. Like this: 
```
Simple Description
value1 <= value2
True: 23 <= 56
```

Reply:
Here is real life example of using those __add_check__
```# Summary:
## Failed checks:
manifest_uploads <= expect_manifest_uploads
-> False: 11815 <= 34

## Passed checks:
actual_byte_rate > produce_byte_rate / throughput_tolerance
-> True: 10082186 > 10485760 / 2

actual_byte_rate < produce_byte_rate * throughput_tolerance
-> True: 10082186 < 10485760 * 2

sum_high_watermarks >= msg_count
-> True: 192000 >= 192000

lag_seconds < config_interval
-> True: 0 < 16

manifest_uploads > 0
-> True: 11815 > 0

segment_uploads > 0
-> True: 9815 > 0
```
We are adding checks, and then evaluate all of them at the end to see ALL results at once along with values calculated. Not one by one with fail on first.